programs/Makefile

```diff
@@ -109,6 +109,7 @@ APPS = \
 	psa/hmac_demo \
 	psa/key_ladder_demo \
 	psa/psa_constant_names \
+	psa/psa_hash \
```

Comment:
There's a directory `programs/psa` that contains all programs that are related to the PSA API: examples of simple usage like `psa_hash` and `crypto_examples` and `aead_demo` and `hmac_demo`, examples of more complex usage like `key_ladder_demo`, and utility programs for developers (just one: `psa_constant_names`). This made sense originally when PSA was an experimental feature, but now that it's the preferred API for cryptography, maybe we shouldn't single out `psa` anymore?

Reply:
Happy to do this as a follow-up PR but in this PR I want to focus on getting the PSA hash program merged.

Reply:
Sure, let's reorganize later.